jobs/build-cosa.Jenkinsfile

```diff
@@ -0,0 +1,145 @@
+def pipeutils, streams, official
+def commit, shortcommit
+node {
+    checkout scm
+    pipeutils = load("utils.groovy")
+    streams = load("streams.groovy")
+    official = pipeutils.isOfficial()
+}
+
+properties([
+    pipelineTriggers([
+        githubPush()
+    ]),
+    parameters([
+      string(name: 'ARCHES',
+             description: 'Space-separated list of target architectures',
+             defaultValue: "x86_64" + " " + streams.additional_arches.join(" "),
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_GIT_URL',
+             description: 'Override the coreos-assembler git repo to use',
+             defaultValue: "https://github.com/coreos/coreos-assembler.git",
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_GIT_REF',
+             description: 'Override the coreos-assembler git ref to use',
+             defaultValue: "",
+             trim: true),
+      string(name: 'CONTAINER_REGISTRY_REPO',
+             description: 'Override the registry to push the container to',
+             defaultValue: "quay.io/coreos-assembler/coreos-assembler",
+             trim: true),
+      string(name: 'CONTAINER_REGISTRY_STAGING_REPO',
+             description: 'Override the staging registry where intermediate images go',
+             defaultValue: "quay.io/coreos-assembler/coreos-assembler-staging",
+             trim: true),
+      string(name: 'COREOS_ASSEMBLER_IMAGE',
+             description: 'Override the coreos-assembler image to use',
+             defaultValue: "coreos-assembler:main",
+             trim: true),
+    ]),
+    buildDiscarder(logRotator(
+        numToKeepStr: '100',
+        artifactNumToKeepStr: '100'
+    )),
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+])
+
+node {
+    change = checkout(
```

Comment:
I think the pseudocode for this would be something like:

```
change = ...
if triggered by git push:
  params.COREOS_ASSEMBLER_GIT_REF = change.GIT_BRANCH[...]
else:
  if params.COREOS_ASSEMBLER_GIT_REF == "":
    params.COREOS_ASSEMBLER_GIT_REF = "main"
  git checkout params.COREOS_ASSEMBLER_GIT_REF
commit=$(git rev-parse HEAD)
```